src/controllers/profileController.ts

```diff
@@ -0,0 +1,85 @@
+import { Request, Response } from 'express';
+import User from '../database/models/user';
+import logger from '../logs/config';
+import uploadImage from '../helpers/claudinary';
+import Role from '../database/models/role';
+import { sendInternalErrorResponse } from '../validations';
+
+export const getUserProfile = async (req: Request, res: Response): Promise<void> => {
+  const userId: string = req.params.userId;
+
+  try {
+    const user = await User.findByPk(userId, {
+      attributes: ['id', 'firstName', 'lastName', 'email', 'phoneNumber', 'photoUrl', 'gender'],
+      include: {
+        model: Role,
+        attributes: ['name'],
+      },
+    });
+
+    if (!user) {
+      res.status(404).json({ message: 'User not found' });
+      return;
+    }
+
+    res.status(200).json({ ok: true, message: user });
+  } catch (error) {
+    logger.error('Error fetching user profile:: ', error);
+    sendInternalErrorResponse(res, error);
+  }
+};
+
+//construct updated fields
+const constructUpdatedFields = (body: any, user: User, uploadedImage?: string) => {
+  return {
+    firstName: body.firstName || user.firstName,
+    lastName: body.lastName || user.lastName,
+    gender: body.gender || user.gender,
+    phoneNumber: body.phoneNumber || user.phoneNumber,
+    photoUrl: uploadedImage || user.photoUrl,
+  };
+};
+
+// Function to construct user response
+const constructUserResponse = (user: User, updatedFields: any) => {
+  return {
+    id: user.id,
+    firstName: updatedFields.firstName,
+    lastName: updatedFields.lastName,
+    email: user.email,
+    phoneNumber: updatedFields.phoneNumber,
+    gender: updatedFields.gender,
+    photoUrl: updatedFields.photoUrl,
+  };
+};
```

Comment:
guess this should be in helpers directory

Reply:
Let us leave it like that so far